source/core/lighting/photons.cpp

```diff
@@ -2280,18 +2280,21 @@ void PhotonGatherer::gatherPhotonsRec(int start, int end)
 
     // check this photon
 
-    // find distance from pt
-    ptToPhoton = - (*pt_s) + Vector3d(photon->Loc);
-    // all distances are squared
-    dx = ptToPhoton[X]*ptToPhoton[X];
-    dy = ptToPhoton[Y]*ptToPhoton[Y];
-    dz = ptToPhoton[Z]*ptToPhoton[Z];
+    // find distance in DimToUse from pt
+    d = photon->Loc[DimToUse] - (*pt_s)[DimToUse];
+    d *= d;
```

Comment:
For clarity, I would recommend to write the above two statements as a single one, making use of the `Sqr()` function defined in `coretypes.h`.

Also for clarity, I would recommend to make liberal use of temporary variables with more descriptive names, e.g. `dSqr` for the value in question. Modern compilers can easily fold multiple local variables into a single memory location if their _effective_ scope doesn't overlap.